src/main/java/com/simplymadeapps/simple_logger_android/SimpleAmazonLogs.java

```diff
@@ -58,73 +58,59 @@ public static void init(Application application) {
 
     // Constructor
     protected SimpleAmazonLogs(Application application) {
-        this.database = Room.databaseBuilder(application, RecordedLogDatabase.class, "recorded_log_database").build();
+        this.storage = SimpleAmazonLogsHelper.newStorageInstance(application.getApplicationContext());
         this.context = application.getApplicationContext();
         this.preferences = PreferenceManager.getDefaultSharedPreferences(context);
         this.editor = preferences.edit();
         this.daysToKeepInStorage = preferences.getInt(KEEP_IN_STORAGE_KEY, 7);
+        this.storagePath = storage.getInternalFilesDirectory()+File.separator+"SIO-Logs";
     }
 
-    /**
-     * @param log takes a string and stores it as a log.  The time stamp will be added automatically.
-     */
-    public static void addLog(final String log) {
-        Thread thread = new Thread() {
-            @Override
-            public void run() {
-                if(haveNotCheckedForOldLogsInLast24Hrs()) { // Only check to clear old logs if we haven't in the last 24 hrs
-                    clearOldLogs();  // Because adding a log is called frequently, we will use it to check for week old logs and delete them
-                }
+    public static void addLog(String log) {
+        if(haveNotCheckedForOldLogsInLast24Hrs()) { // Only check to clear old logs if we haven't in the last 24 hrs
+            clearOldLogs();  // Because adding a log is called frequently, we will use it to check for week old logs and delete them
+        }
 
-                RecordedLog recordedLog = new RecordedLog();
-                recordedLog.setLog(log);
-                recordedLog.setDate(System.currentTimeMillis());
-                database.recordedLogDao().insert(recordedLog);
-            }
-        };
-        thread.start();
+        Date date = new Date(System.currentTimeMillis());
+        String textFileName = date_format_title.format(date);
+        String logDateTag = date_format_log.format(date);
+        storage.createDirectory(storagePath, false);
+        createTextFileIfItDoesntExist(textFileName);
+        storage.appendFile(storagePath+File.separator+textFileName, logDateTag+" - "+log+"");
     }
 
-    /**
-     * @return returns true if we have already checked for old logs today, false if not
-     */
-    protected static boolean haveNotCheckedForOldLogsInLast24Hrs() {
-        return last_clear_old_logs_checked < System.currentTimeMillis()-HRS_24_IN_MS;
+    protected static void createTextFileIfItDoesntExist(String fileName) {
+        boolean exists = storage.isFileExist(storagePath+File.separator+fileName);
```

Comment:
Can we move the filename creation to a helper method rather than append everything inline?